### PR TITLE
Allow visibility specifiers on predicates

### PR DESCRIPTION
**Pull request by emlaufer**
This allows predicates to be marked as public so they can be used across module boundaries, see issue #627 . It enables programs like the following:

```
use prusti_contracts::*;

fn main() {}

mod foo {
    use super::predicate;
    predicate! {
        pub fn ok() -> bool {
            true
        }
    }
}

#[requires(foo::ok())]
fn test_pub_pred() {}
```

### CODE REVIEW

**Comment by dario23** [comment]
thanks for your PR :-)

i'm not actually a maintainer (so i can't approve the CI run, maybe @Aurel300 or @vakaras can help here?), but this looks pretty much exactly like the changes i imagined in #627, so if i didn't miss anything there, i think this should be good to go pretty much immediately.

**Review by vakaras — APPROVED** [review]
LGTM.

**Comment by vakaras** [comment]
Thank you for fixing it!

**Pull request merged**
